### PR TITLE
Fix handling of transactions with explicit isolation level

### PR DESCRIPTION
**Pull request by sds**
First of all, super excited for this plugin! Been wanting to use Kysely with Postgres.js for a while. Thanks for getting it started!

Looks like the current implementation doesn't support transactions with specific isolation levels. Fix that by mirroring what the vanilla Postgres driver does here: https://github.com/kysely-org/kysely/blob/a826cede4007211ec670bfe094de8acdddbbe1cb/src/dialect/postgres/postgres-driver.ts#L58-L66

### CODE REVIEW

**Comment by igalklebanov** [comment]
Hey 👋 

Thanks!

Any reason to prefer `begin` over `start transaction`? I personally prefer being explicit and predictable.

**Comment by sds** [comment]
I tried adding the tests from the upstream Kysely project, but it looks like the `start transaction ...`/`commit` statements don't trigger an event that we can tap into [here](https://github.com/kysely-org/kysely/blob/a826cede4007211ec670bfe094de8acdddbbe1cb/test/node/src/transaction.test.ts#L21).

You're more familiar with the nuts and bolts of Kysely itself—is there a simple workaround?

**Comment by igalklebanov** [comment]
> I tried adding the tests from the upstream Kysely project, but it looks like the `start transaction ...`/`commit` statements don't trigger an event that we can tap into [here](https://github.com/kysely-org/kysely/blob/a826cede4007211ec670bfe094de8acdddbbe1cb/test/node/src/transaction.test.ts#L21).
> 
> You're more familiar with the nuts and bolts of Kysely itself—is there a simple workaround? More specifically: is there a reason we're performing the transaction handling logic in the `PostgresJSConnection` rather than in the `PostgresJSDriver`, like what is done upstream?

Its a hack around Postgres.js's limitations (no single connection getter, transaction API not suitable for Kysely, we have no control over commit/rollback) and how Kysely is built around these parts.

The lifecycle of a transaction in Kysely ~is:
`acquireConnection: connection -> beginTransaction(connection) -> userFunctionality(connection) -> commitTransaction(connection) OR rollbackTransaction(connection) -> releaseConnection(connection)`.

`acquireConnection` is used regardless of transaction scenario, returns a new Connection that has the pool.
If that connection is then used to begin a transaction, we create a new pool of 1 that is used in this Connection until commit/rollback.

I'm OK with leaving this not covered in tests as it requires some hacking to spy on.

File: src/connection.ts [hunk 1]
```diff
@@ -24,9 +24,11 @@ export class PostgresJSConnection implements DatabaseConnection {
...
     this.#transaction = createPostgres({...this.#config, options: {...this.#config.options, max: 1}})
 
-    const statement = `start transaction${isolationLevel ? ` ${isolationLevel}` : ''}`
-
-    await this.#transaction.unsafe(statement)
+    if (isolationLevel) {
+      await this.#transaction.unsafe(`start transaction isolation level ${isolationLevel}`)
+    } else {
+      await this.#transaction.unsafe(`begin`)
+    }
```
Comment (igalklebanov) [inline src/connection.ts:31]: I believe this should allow us to test this like in main repo.

```suggestion
    const compiledQuery = CompiledQuery.raw(
      isolationLevel ? `start transaction isolation level ${isolationLevel}` : 'begin'
    )
    
    await this.executeQuery(compiledQuery)
```

Comment (sds) [reply src/connection.ts]: Nice! Will update.

**Comment by sds** [comment]
Tests pass! Let me know if there's anything else you'd like addressed as part of this PR.

**Review by igalklebanov — APPROVED** [review]
LGTM! 🚀

**Pull request merged**